app/controllers/admins_controller.rb

```diff
@@ -42,11 +33,16 @@ def change_state
   private
 
   def admin_params
-    params[:admin].permit(:name, :email)
+    params.fetch(:admins, {}).permit(:name, :email)
+  end
+
+  def load_admin_or_redirect
+    @admin = Admin.where(id: params[:id]).first
+    redirect_to admins_path if @admin.nil? || @admin.subdomain != request.subdomain
   end
 
-  def name_param
-    params[:admin].permit(:name)
+  def set_admin
```

Comment:
I don't think this method is required. 
It is better to make the definitions of `new` and `create` methods and directly call `Admin.new` in it.